packages/data-point/lib/debug-utils/stack.test.js

```diff
@@ -0,0 +1,439 @@
+      expect(result).toHaveProperty('_stack')
+      expect(result._value).toMatchSnapshot()
+      expect(result._stack).toMatchSnapshot()
+    })
+}
+
+const throwIfEquals = v1 => v2 => {
+  return v1 === v2 ? throwError() : false
+}
+
+nock('http://remote.test')
+  .get('/')
+  .reply(200, {
+    ok: true
+  })
+
+nock('http://remote.test')
+  .get('/source1')
+  .reply(404, 'not found')
+
+const dataPoint = DataPoint.create({
+  entities: {
+    'transform:1': () => {
+      throw new Error('test error')
+    },
+    'transform:2': 'transform:1',
+    'transform:3': 'transform:2',
+
+    'request:1': {
+      value: [identity, throwError],
+      url: 'http://remote.test'
+    },
+    'request:2': {
+      options: {
+        x: () => 'apples',
+        y: [identity, throwError]
+      },
+      url: 'http://remote.test'
+    },
+    'request:3': {
+      before: throwError,
+      url: 'http://remote.test'
+    },
+    'request:4': {
+      url: 'http://remote.test',
+      after: throwError
+    },
+    'request:5': {
+      url: 'http://remote.test/source1'
+    },
+
+    // fail first case
+    'control:1': {
+      select: [
+        { case: throwError, do: throwError },
+        { case: throwError, do: throwError },
+        { default: throwError }
+      ]
+    },
+    // fail second case
+    'control:2': {
+      select: [
+        { case: _false, do: throwError },
+        { case: throwError, do: throwError },
+        { default: throwError }
+      ]
+    },
+    // fail first do
+    'control:3': {
+      select: [
+        { case: _true, do: throwError },
+        { case: throwError, do: throwError },
+        { default: throwError }
+      ]
+    },
+    // fail second do
+    'control:4': {
+      select: [
+        { case: _false, do: throwError },
+        { case: _true, do: throwError },
+        { default: _true }
+      ]
+    },
+    // fail default
+    'control:5': {
+      select: [
+        { case: _false, do: throwError },
+        { case: _false, do: throwError },
+        { default: throwError }
+      ]
+    },
+
+    'model:1': {
+      before: throwError
+    },
+    'model:2': {
+      value: throwError
+    },
+    'model:3': {
+      after: throwError
+    },
+    'model:4': {
+      value: throwError,
+      error: throwError
+    },
+
+    'entry:1': {
+      before: throwError
+    },
+    'entry:2': {
+      value: throwError
+    },
+    'entry:3': {
+      after: throwError
+    },
+    'entry:4': {
+      value: throwError,
+      error: throwError
+    },
+
+    'entry:type-check-1': {
+      inputType: 'string'
+    },
+    'entry:type-check-2': {
+      value: () => 500,
+      outputType: 'string'
+    },
+    'entry:type-check-3': {
+      inputType: 'schema:a.1.0'
+    },
+    'entry:type-check-4': {
+      value: () => 500,
+      outputType: 'schema:a.1.0'
+    },
+
+    'schema:with-value-prop': {
+      value: throwError,
+      schema: {
+        type: 'object'
+      }
+    }
+  }
+})
```

Comment:
I definitely see the value of defining these entities here, but it's not the same pattern as the other entity fixtures we have (in the test directory).

I'm not asking this to be changed or moved, just commenting in case there's interest in discussing the inclusion of this in this file.

Reply:
I like defining them in the same file so it's easier to understand the tests, and since these entities are testing very specific cases, I doubt they would ever be used in other files.

If we do use the same entities in multiple tests (which is encouraged by defining them in the test directory), it could also make it harder to make changes to the entities if we ever need to, since any change would affect multiple tests.

Lastly, if every instance of datapoint that we create for the tests contains all if the entities defined in the test directory, that might slow down the tests (since we probably create a lot of instances).